src/tokens/ERC20Vault.sol

```diff
@@ -0,0 +1,161 @@
+import {SafeTransferLib} from "../utils/SafeTransferLib.sol";
+
+/// @title Yield Bearing Vault
+/// @author joeysantoro, Transmissions11 and JetJadeja
+contract ERC20Vault is ERC20 {
+    using SafeTransferLib for ERC20;
+    using FixedPointMathLib for uint256;
+
+    /*///////////////////////////////////////////////////////////////
+                                IMMUTABLES
+    //////////////////////////////////////////////////////////////*/
+
+    /// @notice The underlying token the Vault accepts.
+    ERC20 public immutable underlying;
+
+    /// @notice The base unit of the underlying token and hence vault.
+    /// @dev Equal to 10 ** decimals. Used for fixed point arithmetic.
+    uint256 public immutable baseUnit;
+
+    /// @notice Creates a new Vault that accepts a specific underlying token.
+    /// @param _underlying The ERC20 compliant token the Vault should accept.
+    /// @param _namePrefix A name prefix to be applied before
+    /// @param _symbolPrefix The ERC20 compliant token the Vault should accept.
+
+    constructor(
+        ERC20 _underlying,
+        string memory _namePrefix,
+        string memory _symbolPrefix
+    )
+        ERC20(
+            string(abi.encodePacked(_namePrefix, _underlying.name(), " Vault")),
+            string(abi.encodePacked(_symbolPrefix, _underlying.symbol())),
+            _underlying.decimals()
+        )
+    {
+        underlying = _underlying;
+
+        baseUnit = 10**decimals;
+
+        // Prevent minting of shares until
+        // the initialize function is called.
+        totalSupply = type(uint256).max;
+    }
+
+    /*///////////////////////////////////////////////////////////////
+                        DEPOSIT/WITHDRAWAL LOGIC
+    //////////////////////////////////////////////////////////////*/
+
+    /// @notice Emitted after a successful deposit.
+    /// @param user The address that deposited into the Vault.
+    /// @param underlyingAmount The amount of underlying tokens that were deposited.
+    event Deposit(address indexed user, uint256 underlyingAmount);
+
+    /// @notice Emitted after a successful withdrawal.
+    /// @param owner The address that withdrew from the Vault.
+    /// @param to The destination for withdrawn tokens.
+    /// @param underlyingAmount The amount of underlying tokens that were withdrawn.
+    event Withdraw(address indexed owner, address indexed to, uint256 underlyingAmount);
+
+    /// @notice Deposit a specific amount of underlying tokens.
+    /// @param to The address to receive shares corresponding to the deposit
+    /// @param underlyingAmount The amount of the underlying token to deposit.
+    function deposit(address to, uint256 underlyingAmount) external virtual returns (uint256 shares) {
+        // We don't allow depositing 0 to prevent emitting a useless event.
+        require(underlyingAmount != 0, "AMOUNT_CANNOT_BE_ZERO");
+
+        shares = underlyingAmount.fdiv(exchangeRate(), baseUnit);
```

Comment:
lez do this inline like `_mint(to, shares = underlyingAmount.fdiv(exchangeRate(), baseUnit));`

Reply:
just out of curiosity (I'm totally new to optimizations and I'm lurking hard on your Twitter timeline :D): with the inline version, you are losing readability by a lot. inlining things let you gain optimization or it's just more of a "style"?

Am I being to naive to think that the bytecode produced is optimized automatically with the 2 line version?

Reply:
was thinking about style here, tho in some cases iirc i've found inlining cheaper

Reply:
if its a style thing I actually think its more readable outside